### PR TITLE
Elasticsearch: explicitly require `--db-user`.

### PR DESCRIPTION
**Pull request by Tener**
Currently Elasticsearch requires `--db-user` or an error will be thrown on connection attempt:

```
$ tsh proxy db  myelastic --port 3333 --tunnel
Started authenticated tunnel for the Elasticsearch database "myelastic" in cluster "boson.tener.io" on 127.0.0.1:3333.

Use the following command to connect to the database or to the address above using other database GUI/CLI clients:
  $ curl http://localhost:3333/
```

```
$ curl http://localhost:3333/
{"reason":"missing parameter Subject.Common name","type":"internal_server_error_exception"}%
```

```
023-01-25T16:11:10+01:00 ERRO [DB:SERVIC] Failed to handle connection. addr:100.74.253.103:3080 error:[
ERROR REPORT:
Original Error: *trace.BadParameterError missing parameter Subject.Common name
Stack Trace:
	github.com/gravitational/teleport/api@v0.0.0/client/client.go:1467 github.com/gravitational/teleport/api/client.(*Client).GenerateDatabaseCert
	github.com/gravitational/teleport/lib/srv/db/common/auth.go:703 github.com/gravitational/teleport/lib/srv/db/common.(*dbAuth).getClientCert
	github.com/gravitational/teleport/lib/srv/db/common/auth.go:545 github.com/gravitational/teleport/lib/srv/db/common.(*dbAuth).appendClientCert
	github.com/gravitational/teleport/lib/srv/db/common/auth.go:504 github.com/gravitational/teleport/lib/srv/db/common.(*dbAuth).getTLSConfigVerifyFull
	github.com/gravitational/teleport/lib/srv/db/common/auth.go:444 github.com/gravitational/teleport/lib/srv/db/common.(*dbAuth).GetTLSConfig
	github.com/gravitational/teleport/lib/srv/db/elasticsearch/engine.go:121 github.com/gravitational/teleport/lib/srv/db/elasticsearch.(*Engine).HandleConnection
	github.com/gravitational/teleport/lib/srv/db/server.go:905 github.com/gravitational/teleport/lib/srv/db.(*Server).handleConnection
	github.com/gravitational/teleport/lib/srv/db/server.go:815 github.com/gravitational/teleport/lib/srv/db.(*Server).HandleConnection
	github.com/gravitational/teleport/lib/reversetunnel/transport.go:275 github.com/gravitational/teleport/lib/reversetunnel.(*transport).start
	github.com/gravitational/teleport/lib/reversetunnel/agent.go:581 github.com/gravitational/teleport/lib/reversetunnel.(*agent).handleDrainChannels.func2
	runtime/asm_amd64.s:1594 runtime.goexit
User Message: missing parameter Subject.Common name] db/server.go:817
```

This change improves the UX by making the requirement explicit for `tsh`. An explicit check is added server-side to accommodate old clients which may not have the check implemented client-side.

### CODE REVIEW

File: lib/srv/db/elasticsearch/engine.go [hunk 1]
```diff
@@ -118,6 +118,10 @@ func (e *Engine) HandleConnection(ctx context.Context, sessionCtx *common.Sessio
 
 	clientConnReader := bufio.NewReader(e.clientConn)
 
+	if sessionCtx.Identity.RouteToDatabase.Username == "" {
+		return trace.BadParameter("database username required")
```
Comment (zmb3) [inline lib/srv/db/elasticsearch/engine.go:122]: Is it worth mentioning elasticsearch specifically in the error message? This way we don't lead the user to think the username is always required.

File: tool/tsh/db.go [hunk 2]
```diff
@@ -728,6 +729,9 @@ func onDatabaseConnect(cf *CLIConf) error {
 	if routeToDatabase.Protocol == defaults.ProtocolDynamoDB {
 		return trace.BadParameter(formatDbCmdUnsupportedDBProtocol(cf, routeToDatabase))
 	}
+	// if routeToDatabase.Protocol == defaults.ProtocolElasticsearch && routeToDatabase.Username == "" {
```
Comment (zmb3) [inline tool/tsh/db.go:732]: If this is not necessary then lets remove it. No need to keep commented-out code.

Comment (Tener) [reply tool/tsh/db.go:732]: Oops, my bad. This wasn't meant to be here.

File: lib/srv/db/elasticsearch/engine.go (hunk 1 above)
Comment (Tener) [reply lib/srv/db/elasticsearch/engine.go:122]: Sure, we can have that.

**Review by Joerger — APPROVED** [review]

**Review by smallinsky — APPROVED** [review]

**Pull request merged**